From 039bd396a88b93edd0008cab8f0b2e597ec072cc Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 6 May 2022 09:18:15 +0800
Subject: [PATCH] sw64: remove unused header files
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 039bd396a88b93edd0008cab8f0b2e597ec072cc
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

These header files have became unused for a long time.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/agp.h            |  19 ----
 arch/sw_64/include/asm/asm-prototypes.h |   1 -
 arch/sw_64/include/asm/compiler.h       |   7 --
 arch/sw_64/include/asm/console.h        |  11 ---
 arch/sw_64/include/asm/elf.h            |   6 +-
 arch/sw_64/include/asm/floppy.h         | 116 ------------------------
 arch/sw_64/include/asm/special_insns.h  |  20 ----
 arch/sw_64/include/uapi/asm/console.h   |  51 -----------
 8 files changed, 2 insertions(+), 229 deletions(-)
 delete mode 100644 arch/sw_64/include/asm/agp.h
 delete mode 100644 arch/sw_64/include/asm/compiler.h
 delete mode 100644 arch/sw_64/include/asm/console.h
 delete mode 100644 arch/sw_64/include/asm/floppy.h
 delete mode 100644 arch/sw_64/include/asm/special_insns.h
 delete mode 100644 arch/sw_64/include/uapi/asm/console.h

diff --git a/arch/sw_64/include/asm/agp.h b/arch/sw_64/include/asm/agp.h
deleted file mode 100644
index e9d16888910e..000000000000
--- a/arch/sw_64/include/asm/agp.h
+++ /dev/null
@@ -1,19 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _ASM_SW64_AGP_H
-#define _ASM_SW64_AGP_H 1
-
-#include <asm/io.h>
-
-/* dummy for now */
-
-#define map_page_into_agp(page)
-#define unmap_page_from_agp(page)
-#define flush_agp_cache() mb()
-
-/* GATT allocation. Returns/accepts GATT kernel virtual address. */
-#define alloc_gatt_pages(order)		\
-	((char *)__get_free_pages(GFP_KERNEL, (order)))
-#define free_gatt_pages(table, order)	\
-	free_pages((unsigned long)(table), (order))
-
-#endif
diff --git a/arch/sw_64/include/asm/asm-prototypes.h b/arch/sw_64/include/asm/asm-prototypes.h
index 21f4f494d74d..15bad8ef6883 100644
--- a/arch/sw_64/include/asm/asm-prototypes.h
+++ b/arch/sw_64/include/asm/asm-prototypes.h
@@ -4,7 +4,6 @@
 
 #include <linux/spinlock.h>
 #include <asm/checksum.h>
-#include <asm/console.h>
 #include <asm/page.h>
 #include <asm/string.h>
 #include <linux/uaccess.h>
diff --git a/arch/sw_64/include/asm/compiler.h b/arch/sw_64/include/asm/compiler.h
deleted file mode 100644
index 9a80aa6a0ba8..000000000000
--- a/arch/sw_64/include/asm/compiler.h
+++ /dev/null
@@ -1,7 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _ASM_SW64_COMPILER_H
-#define _ASM_SW64_COMPILER_H
-
-#include <uapi/asm/compiler.h>
-
-#endif /* _ASM_SW64_COMPILER_H */
diff --git a/arch/sw_64/include/asm/console.h b/arch/sw_64/include/asm/console.h
deleted file mode 100644
index 0c01cb740bce..000000000000
--- a/arch/sw_64/include/asm/console.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _ASM_SW64_CONSOLE_H
-#define _ASM_SW64_CONSOLE_H
-
-#include <uapi/asm/console.h>
-#ifndef __ASSEMBLY__
-struct crb_struct;
-extern int callback_init_done;
-extern void callback_init(void);
-#endif /* __ASSEMBLY__ */
-#endif /* _ASM_SW64_CONSOLE_H */
diff --git a/arch/sw_64/include/asm/elf.h b/arch/sw_64/include/asm/elf.h
index 150629b0b615..abecc5653eba 100644
--- a/arch/sw_64/include/asm/elf.h
+++ b/arch/sw_64/include/asm/elf.h
@@ -3,7 +3,6 @@
 #define _ASM_SW64_ELF_H
 #ifdef __KERNEL__
 #include <asm/auxvec.h>
-#include <asm/special_insns.h>
 #endif
 /* Special values for the st_other field in the symbol table.  */
 
@@ -141,11 +140,10 @@ extern int dump_elf_task_fp(elf_fpreg_t *dest, struct task_struct *task);
 
 /*
  * This yields a mask that user programs can use to figure out what
- * instruction set this CPU supports.  This is trivial on SW-64,
- * but not so on other machines.
+ * instruction set this CPU supports.
  */
 
-#define ELF_HWCAP  (~amask(-1))
+#define ELF_HWCAP	0
 
 /*
  * This yields a string that ld.so will use to load implementation
diff --git a/arch/sw_64/include/asm/floppy.h b/arch/sw_64/include/asm/floppy.h
deleted file mode 100644
index f4646d99d80c..000000000000
--- a/arch/sw_64/include/asm/floppy.h
+++ /dev/null
@@ -1,116 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-/*
- * Architecture specific parts of the Floppy driver
- *
- * This file is subject to the terms and conditions of the GNU General Public
- * License.  See the file "COPYING" in the main directory of this archive
- * for more details.
- *
- * Copyright (C) 1995
- */
-#ifndef _ASM_SW64_FLOPPY_H
-#define _ASM_SW64_FLOPPY_H
-
-#define fd_inb(port)		inb_p(port)
-#define fd_outb(value, port)	outb_p(value, port)
-
-#define fd_enable_dma()		enable_dma(FLOPPY_DMA)
-#define fd_disable_dma()	disable_dma(FLOPPY_DMA)
-#define fd_request_dma()	request_dma(FLOPPY_DMA, "floppy")
-#define fd_free_dma()		free_dma(FLOPPY_DMA)
-#define fd_clear_dma_ff()	clear_dma_ff(FLOPPY_DMA)
-#define fd_set_dma_mode(mode)	set_dma_mode(FLOPPY_DMA, mode)
-#define fd_set_dma_addr(addr)	set_dma_addr(FLOPPY_DMA, virt_to_bus(addr))
-#define fd_set_dma_count(count)	set_dma_count(FLOPPY_DMA, count)
-#define fd_enable_irq()		enable_irq(FLOPPY_IRQ)
-#define fd_disable_irq()	disable_irq(FLOPPY_IRQ)
-#define fd_cacheflush(addr, size) /* nothing */
-#define fd_request_irq() \
-	request_irq(FLOPPY_IRQ, floppy_interrupt, 0, "floppy", NULL)
-#define fd_free_irq()		free_irq(FLOPPY_IRQ, NULL)
-
-#ifdef CONFIG_PCI
-
-#include <linux/pci.h>
-
-#define fd_dma_setup(addr, size, mode, io) \
-	sw64_fd_dma_setup(addr, size, mode, io)
-
-static inline int
-sw64_fd_dma_setup(char *addr, unsigned long size, int mode, int io)
-{
-	static unsigned long prev_size;
-	static dma_addr_t bus_addr;
-	static char *prev_addr;
-	static int prev_dir;
-	int dir;
-
-	dir = (mode != DMA_MODE_READ) ? PCI_DMA_FROMDEVICE : PCI_DMA_TODEVICE;
-
-	if (bus_addr
-		&& (addr != prev_addr || size != prev_size || dir != prev_dir)) {
-		/* different from last time -- unmap prev */
-		bus_addr = 0;
-	}
-
-	if (!bus_addr)	/* need to map it */
-		bus_addr = virt_to_bus(addr);
-
-	/* remember this one as prev */
-	prev_addr = addr;
-	prev_size = size;
-	prev_dir = dir;
-
-	fd_clear_dma_ff();
-	fd_cacheflush(addr, size);
-	fd_set_dma_mode(mode);
-	set_dma_addr(FLOPPY_DMA, bus_addr);
-	fd_set_dma_count(size);
-	virtual_dma_port = io;
-	fd_enable_dma();
-
-	return 0;
-}
-
-#endif /* CONFIG_PCI */
-
-inline void virtual_dma_init(void)
-{
-	/* Nothing to do on an sw64 */
-}
-
-static int FDC1 = 0x3f0;
-static int FDC2 = -1;
-
-/*
- * Again, the CMOS information doesn't work on the sw64..
- */
-#define FLOPPY0_TYPE 6
-#define FLOPPY1_TYPE 0
-
-#define N_FDC 2
-#define N_DRIVE 8
-
-/*
- * Most sw64s have no problems with floppy DMA crossing 64k borders,
- * except for certain ones, like XL and RUFFIAN.
- *
- * However, the test is simple and fast, and this *is* floppy, after all,
- * so we do it for all platforms, just to make sure.
- *
- * This is advantageous in other circumstances as well, as in moving
- * about the PCI DMA windows and forcing the floppy to start doing
- * scatter-gather when it never had before, and there *is* a problem
- * on that platform... ;-}
- */
-
-static inline unsigned long CROSS_64KB(void *a, unsigned long s)
-{
-	unsigned long p = (unsigned long)a;
-
-	return ((p + s - 1) ^ p) & ~0xffffUL;
-}
-
-#define EXTRA_FLOPPY_PARAMS
-
-#endif /* __ASM_SW64_FLOPPY_H */
diff --git a/arch/sw_64/include/asm/special_insns.h b/arch/sw_64/include/asm/special_insns.h
deleted file mode 100644
index 7f5a52b20444..000000000000
--- a/arch/sw_64/include/asm/special_insns.h
+++ /dev/null
@@ -1,20 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _ASM_SW64_SPECIAL_INSNS_H
-#define _ASM_SW64_SPECIAL_INSNS_H
-
-enum amask_enum {
-	AMASK_BWX = (1UL << 0),
-	AMASK_FIX = (1UL << 1),
-	AMASK_CIX = (1UL << 2),
-	AMASK_MAX = (1UL << 8),
-	AMASK_PRECISE_TRAP = (1UL << 9),
-};
-
-#define amask(mask)						\
-({								\
-	unsigned long __amask, __input = (mask);		\
-	__asm__ ("mov %1, %0" : "=r"(__amask) : "rI"(__input));	\
-	__amask;						\
-})
-
-#endif /* _ASM_SW64_SPECIAL_INSNS_H */
diff --git a/arch/sw_64/include/uapi/asm/console.h b/arch/sw_64/include/uapi/asm/console.h
deleted file mode 100644
index a40cd7aeb31f..000000000000
--- a/arch/sw_64/include/uapi/asm/console.h
+++ /dev/null
@@ -1,51 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
-#ifndef _UAPI_ASM_SW64_CONSOLE_H
-#define _UAPI_ASM_SW64_CONSOLE_H
-
-/*
- * Console callback routine numbers
- */
-#define CCB_GETC		0x01
-#define CCB_PUTS		0x02
-#define CCB_RESET_TERM		0x03
-#define CCB_SET_TERM_INT	0x04
-#define CCB_SET_TERM_CTL	0x05
-#define CCB_PROCESS_KEYCODE	0x06
-#define CCB_OPEN_CONSOLE	0x07
-#define CCB_CLOSE_CONSOLE	0x08
-
-#define CCB_OPEN		0x10
-#define CCB_CLOSE		0x11
-#define CCB_IOCTL		0x12
-#define CCB_READ		0x13
-#define CCB_WRITE		0x14
-
-#define CCB_SET_ENV		0x20
-#define CCB_RESET_ENV		0x21
-#define CCB_GET_ENV		0x22
-#define CCB_SAVE_ENV		0x23
-
-#define CCB_PSWITCH		0x30
-#define CCB_BIOS_EMUL		0x32
-
-/*
- * Environment variable numbers
- */
-#define ENV_AUTO_ACTION		0x01
-#define ENV_BOOT_DEV		0x02
-#define ENV_BOOTDEF_DEV		0x03
-#define ENV_BOOTED_DEV		0x04
-#define ENV_BOOT_FILE		0x05
-#define ENV_BOOTED_FILE		0x06
-#define ENV_BOOT_OSFLAGS	0x07
-#define ENV_BOOTED_OSFLAGS	0x08
-#define ENV_BOOT_RESET		0x09
-#define ENV_DUMP_DEV		0x0A
-#define ENV_ENABLE_AUDIT	0x0B
-#define ENV_LICENSE		0x0C
-#define ENV_CHAR_SET		0x0D
-#define ENV_LANGUAGE		0x0E
-#define ENV_TTY_DEV		0x0F
-
-
-#endif /* _UAPI_ASM_SW64_CONSOLE_H */
-- 
2.33.0

